### PR TITLE
Fix Credits Dark Mode Issues and refactor webview code

### PR DESCRIPTION
**Pull request by hilmyveradin**
To fix this issue:

https://github.com/OneBusAway/onebusaway-ios/issues/720

I created this PR with this result:

Simulator Screen Recording - iPhone 15 Pro - 2024-03-26 at 22 02 10

### CODE REVIEW

File: OBAKit/WebView/DocumentWebView.swift [hunk 1]
```diff
@@ -25,20 +25,23 @@ class DocumentWebView: WKWebView {
...
-            content = content.replacingOccurrences(of: "{{{oba_page_actions}}}", with: buttonText)
-        }
+        var content = pageBody
+            .replacingOccurrences(of: "{{{oba_page_content}}}", with: htmlFragment)
+            .replacingOccurrences(of: "{{{accent_color}}}", with: accentHexColor)
+            .replacingOccurrences(of: "{{{accent_foreground_color}}}", with: accentForegroundColor)
+
+        // Construct button HTML if actionButtonTitle is present and not empty
+        let buttonText: String? = actionButtonTitle.flatMap { title in
+            !title.isEmpty ? """
```
Comment (aaronbrethorst) [inline OBAKit/WebView/DocumentWebView.swift:35]: I'd prefer seeing this expressed as `title.isEmpty ? nil : """ <html content goes here> """`

* https://medium.com/@Cuadraman/why-to-stop-writting-negavite-code-af5ffb17195
* https://softwareengineering.stackexchange.com/questions/350472/developer-insists-if-statements-shouldnt-have-negated-conditions-and-should-al

Comment (hilmyveradin) [reply OBAKit/WebView/DocumentWebView.swift:35]: Understood. Will fix this now!

**Review by aaronbrethorst — CHANGES_REQUESTED** [review]
code changes look good. please squash the three commits into a single one.

**Review by aaronbrethorst — APPROVED** [review]
👍

**Pull request merged**